include/crocoddyl/multibody/costs/contact-wrench-cone.hxx

```diff
@@ -0,0 +1,125 @@
+    throw_pretty("Invalid argument: "
+                 << "nr is equals to " << fref_.oRf.get_nf() + 1);
+  }
+}
+
+template <typename Scalar>
+CostModelContactWrenchConeTpl<Scalar>::CostModelContactWrenchConeTpl(
+    boost::shared_ptr<StateMultibody> state, boost::shared_ptr<ActivationModelAbstract> activation,
+    const FrameWrenchCone& fref)
+    : Base(state, activation), fref_(fref) {
+  if (activation_->get_nr() != fref_.oRf.get_nf() + 1) {
+    throw_pretty("Invalid argument: "
+                 << "nr is equals to " << fref_.oRf.get_nf() + 1);
+  }
+}
+
+template <typename Scalar>
+CostModelContactWrenchConeTpl<Scalar>::CostModelContactWrenchConeTpl(boost::shared_ptr<StateMultibody> state,
+                                                                         const FrameWrenchCone& fref,
+                                                                         const std::size_t& nu)
+    : Base(state, fref.oRf.get_nf() + 1, nu), fref_(fref) {}
+
+template <typename Scalar>
+CostModelContactWrenchConeTpl<Scalar>::CostModelContactWrenchConeTpl(boost::shared_ptr<StateMultibody> state,
+                                                                         const FrameWrenchCone& fref)
+    : Base(state, fref.oRf.get_nf() + 1), fref_(fref) {}
+
+template <typename Scalar>
+CostModelContactWrenchConeTpl<Scalar>::~CostModelContactWrenchConeTpl() {}
+
+template <typename Scalar>
+void CostModelContactWrenchConeTpl<Scalar>::calc(const boost::shared_ptr<CostDataAbstract>& data,
+                                                   const Eigen::Ref<const VectorXs>&,
+                                                   const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+
+  // Compute the residual of the wrench cone. Note that we need to transform the wrench
+  // to the contact frame
+  data->r.noalias() = fref_.oRf.get_A() * d->contact->jMf.actInv(d->contact->f).toVector();
+
+  // Compute the cost
+  activation_->calc(data->activation, data->r);
+  data->cost = data->activation->a_value;
+}
+
+template <typename Scalar>
+void CostModelContactWrenchConeTpl<Scalar>::calcDiff(const boost::shared_ptr<CostDataAbstract>& data,
+                                                       const Eigen::Ref<const VectorXs>&,
+                                                       const Eigen::Ref<const VectorXs>&) {
+  Data* d = static_cast<Data*>(data.get());
+
+  const MatrixXs& df_dx = d->contact->df_dx;
+  const MatrixXs& df_du = d->contact->df_du;
+  const MatrixX6s& A = fref_.oRf.get_A();
+
+  activation_->calcDiff(data->activation, data->r);
+  
+  data->Rx.noalias() = A * df_dx;
+  data->Ru.noalias() = A * df_du;
+
+  data->Lx.noalias() = data->Rx.transpose() * data->activation->Ar;
+  data->Lu.noalias() = data->Ru.transpose() * data->activation->Ar;
+
+  d->Arr_Ru.noalias() = data->activation->Arr * data->Ru;
+
+  data->Lxx.noalias() = data->Rx.transpose() * data->activation->Arr * data->Rx;
```

Comment:
Let's use a third variable with the name of `Arr_Rx`, i.e.
```cpp
d->Arr_Rx.noalias() = data->activation->Arr * data->Rx
```

This code will prevent Eigen to allocate data. The command `noalias()` only prevents data allocation for two matrices product. When we have three (or more) matrices multiplications, the only way to prevent memory allocation is by defining extra variables which we allocate once.

To do this, you will need to define this data object (```Arr_Rx```) in the data class of this cost function.

Additionally, I noticed that this inefficiency appears the in the friction cone cost. We need to modify it as well, however, you don't have to do it in this PR if you're busy.